src/main/java/edu/ksu/canvas/attendance/services/AttendanceSectionService.java

```diff
@@ -25,9 +34,110 @@ public AttendanceSection getFirstSectionOfCourse(long canvasCourseId) {
         return sections.isEmpty() ? null : sections.get(0);
     }
 
+    public List<AttendanceSection> getSectionByCanvasCourseId(long canvasCourseId) {
+        return sectionRepository.findByCanvasCourseId(canvasCourseId);
+    }
 
     public List<AttendanceSection> getSectionsByCourse(long canvasCourseId) {
         return sectionRepository.findByCanvasCourseId(canvasCourseId);
     }
 
+    /**
+     * @throws RuntimeException when courseForm is null
+     */
+    public void save(CourseConfigurationForm courseForm, long canvasCourseId) {
+        Validate.notNull(courseForm, "courseForm must not be null");
+
+        List<AttendanceSection> sections = sectionRepository.findByCanvasCourseId(canvasCourseId);
+        if(sections == null || sections.isEmpty()) {
+            RuntimeException e = new IllegalArgumentException("Cannot load data into courseForm for non-existant course");
```

Comment:
This message does not describe the situation. The course will exist at this point even if this error is reached.